lasagne/layers/merge.py

```diff
@@ -11,6 +17,217 @@
 ]
 
 
+CROP_NONE = None
+CROP_LOWER = 'lower'
+CROP_CENTER = 'center'
+CROP_UPPER = 'upper'
+
+
+def autocrop(inputs, cropping):
+    """
+    Crops the given input arrays.
+
+    Cropping takes a sequence of inputs and crops them per-axis in order to
+    ensure that their sizes are consistent so that they can be combined
+    in an element-wise fashion. If cropping is enabled for a specific axis,
+    the minimum size in that axis of all inputs is computed, and all
+    inputs are cropped to that size.
+
+    The per-axis cropping modes are:
+
+    `CROP_NONE` (`None`): this axis is not cropped, inputs are unchanged in
+    this axis
+
+    `CROP_LOWER` (`'lower'`): inputs are cropped choosing the lower portion
+    in this axis (`a[:crop_size, ...]`)
+
+    `CROP_UPPER` (`'lower'`): inputs are cropped choosing the upper portion
+    in this axis (`a[-crop_size:, ...]`)
+
+    `CROP_CENTER` (`'center'`): inputs are cropped choosing the central
+    portion in this axis (`a[offset:offset+crop_size, ...]` where
+    `offset = (a.shape[0]-crop_size)//2)
+
+    Parameters
+    ----------
+    inputs : the input arrays in the form of a list of Theano expressions
+
+    cropping : a list of cropping modes, one for each axis. If length of
+        `cropping` is less than the number of axes in the inputs, it is
+        padded with `CROP_NONE`. If `cropping` is None, `input_shapes`
+        is returned as is.
+
+    Returns
+    -------
+    list of Theano expressions
+        each expression is the cropped version of the corresponding input
+
+    Example
+    -------
+    For example, given three inputs:
+
+    >>> import numpy
+    >>> import theano
+    >>> from mock import Mock
+
+    >>> a = numpy.random.random((1, 2, 3, 4))
+    >>> b = numpy.random.random((5, 4, 4, 2))
+    >>> c = numpy.random.random((7, 1, 8, 9))
+
+    Cropping mode for each axis:
+
+    >>> cropping = [CROP_NONE, CROP_LOWER, CROP_CENTER, CROP_UPPER]
+
+    Crop (note that the input arrays are converted to Theano vars first,
+    and that the results are converted back from Theano expressions to
+    numpy arrays by calling `eval()`)
+    >>> xa, xb, xc = autocrop([theano.shared(a), \
+                               theano.shared(b), \
+                               theano.shared(c)], cropping)
+    >>> xa, xb, xc = xa.eval(), xb.eval(), xc.eval()
+
+    They will be left as is in axis 0 and cropped in the other three,
+    choosing the lower, center and upper portions:
+
+    Axis 0: choose all, axis 1: lower 1 element,
+    axis 2: central 3 (all) and axis 3: upper 2
+    >>> assert (xa == a[:, :1, :3, -2:]).all()
```

Comment:
Instead of an assert, you can also write:
```
>>> (x< == a[:, :1, :3, -2:]).all()
True
```
That one simulates entering something into a prompt and getting back the result. doctests will then check whether it gets the expected result.